From 0b016c36d8206ac4513bee36ee8a420ce8c14cde Mon Sep 17 00:00:00 2001
From: Arie Brown <abrown@habana.ai>
Date: Thu, 04 Mar 2021 00:24:11 +0200
Subject: [PATCH] [SW-31828] gaudi: update bar4 size

Since security introduced the 1GB window cannot be changed by LKD to
scan entire 32GB HBM, so the size of BAR4 updated to 32GB

Change-Id: I59a6c539be161f595ca9a43ab3c9360fbf68a3e6
Signed-off-by: Arie Brown <abrown@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index b3d9e20..90bf522 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -160,6 +160,11 @@
 	regmap_writel(0x0004A07D, mmPCIE_PHY_CR_PARA_CMD);
 	udelay(20);
 
+	regmap_writel(0x00100000, mmPCIE_AUX_DBI);
+	regmap_writel(0xFFFFFFFF, mmPCIE_DBI_BAR4_REG);
+	regmap_writel(0x00000007, mmPCIE_DBI_BAR5_REG);
+	regmap_writel(0x00000000, mmPCIE_AUX_DBI);
+
 	/* Enable and Mask LTSSM (erratum 102) */
 	regmap_writel(3, mmPCIE_AUX_LTSSM_EN);
 }
